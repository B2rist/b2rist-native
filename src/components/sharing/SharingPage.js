import React, {useCallback, useMemo} from 'react';
import {StyleSheet, View} from 'react-native';
import {Button, FAB} from 'react-native-paper';
import {matchRoutes, useLocation, useNavigate} from 'react-router-native';
import BackgroundScrollView from 'components/custom/BackgroundScrollView';
import {useTranslation} from 'react-i18next';
import routes from 'routes';
import SharingPoints from 'components/sharing/SharingPoints';
import SharingRoutes from 'components/sharing/SharingRoutes';

const routeElements = {
  ROUTES: 'ROUTES',
  POINTS: 'POINTS',
};

const selectedElementMap = {
  '/sharing/routes': routeElements.ROUTES,
  '/sharing/points': routeElements.POINTS,
};

const sharingPageResolver = {
  [routeElements.ROUTES]: <SharingRoutes />,
  [routeElements.POINTS]: <SharingPoints />,
};

const createElementLinks = {
  [routeElements.ROUTES]: '/route',
  [routeElements.POINTS]: '/point',
};

const SharingPage = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const {t} = useTranslation();
  const [{route}] = matchRoutes(routes, location);

  const selectedElement = useMemo(
    () => selectedElementMap[route.path],
    [route.path],
  );

  const createElement = useCallback(() => {
    navigate(createElementLinks[selectedElement]);
  }, [navigate, selectedElement]);

  const navigateRoutes = useCallback(() => {
    navigate('/sharing/routes');
  }, [navigate]);

  const navigatePoints = useCallback(() => {
    navigate('/sharing/points');
  }, [navigate]);

  return (
    <View style={styles.page}>
      <BackgroundScrollView>
        <View style={styles.elementContainer}>
          <Button
            style={styles.elementButton}
            icon="navigation-variant"
            mode={
              selectedElement === routeElements.ROUTES
                ? 'contained'
                : 'elevated'
            }
            onPress={navigateRoutes}>
            {t('mainMenu.routes')}
          </Button>
          <Button
            style={styles.elementButton}
            icon="map-marker"
            mode={
              selectedElement === routeElements.POINTS
                ? 'contained'
                : 'elevated'
            }
            onPress={navigatePoints}>
            {t('mainMenu.points')}
          </Button>
        </View>
        <View style={styles.content}>
          {sharingPageResolver[selectedElement]}
        </View>
      </BackgroundScrollView>
      <FAB icon="plus" style={styles.fab} onPress={createElement} />
    </View>
  );
};

const styles = StyleSheet.create({
  page: {
    flex: 1,
  },
  elementContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  elementButton: {
    margin: 5,
    flexBasis: '46%',
  },
  content: {
    flex: 1,
    flexWrap: 'wrap',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  fab: {
    position: 'absolute',
    margin: 16,
    right: 0,
    borderRadius: 50,
    bottom: 0,
  },
});

export default SharingPage;
