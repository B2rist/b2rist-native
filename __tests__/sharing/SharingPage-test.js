import 'react-native';
import React from 'react';
import SharingPage from 'components/sharing/SharingPage';
import renderer from 'react-test-renderer';

jest.mock('react-router-native', () => ({
  ...jest.requireActual('react-router-native'),
  useLocation: () => ({pathname: '/sharing/points'}),
  useNavigate: jest.fn(),
}));

jest.mock('react-redux', () => {
  const ActualRedux = jest.requireActual('react-redux');
  return {
    ...ActualRedux,
    useSelector: () => ({list: [], loading: false}),
  };
});

const wrapper = () => renderer.create(<SharingPage />);

describe('sharing page tests', () => {
  it('sharing page renders test', () => {
    expect(wrapper().toJSON()).toMatchSnapshot();
  });
});
