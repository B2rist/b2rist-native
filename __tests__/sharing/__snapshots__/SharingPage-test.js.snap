// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sharing page tests sharing page renders test 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      {
        "backgroundColor": "#F3F3F3",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      contentInsetAdjustmentBehavior="automatic"
      style={
        {
          "backgroundColor": "#F3F3F3",
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignSelf": undefined,
                "bottom": undefined,
                "end": undefined,
                "left": undefined,
                "position": undefined,
                "right": undefined,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 3,
                "start": undefined,
                "top": undefined,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "rgb(247, 243, 249)",
                    "borderColor": "transparent",
                    "borderRadius": 20,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "flexBasis": "46%",
                    "margin": 5,
                    "minWidth": 64,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      false,
                      {
                        "borderRadius": 20,
                      },
                    ]
                  }
                  testID="button"
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "marginLeft": 12,
                            "marginRight": -4,
                          },
                          {
                            "marginLeft": 16,
                            "marginRight": -16,
                          },
                          false,
                        ]
                      }
                      testID="button-icon-container"
                    >
                      <Text
                        accessibilityElementsHidden={true}
                        allowFontScaling={false}
                        importantForAccessibility="no-hide-descendants"
                        pointerEvents="none"
                        selectable={false}
                        style={
                          [
                            {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontSize": 18,
                            },
                            [
                              {
                                "lineHeight": 18,
                                "transform": [
                                  {
                                    "scaleX": 1,
                                  },
                                ],
                              },
                              {
                                "backgroundColor": "transparent",
                              },
                            ],
                            {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                      >
                        ó±£°
                      </Text>
                    </View>
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          {
                            "textAlign": "left",
                          },
                          {
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      mainMenu.routes
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "alignSelf": undefined,
                "bottom": undefined,
                "end": undefined,
                "left": undefined,
                "position": undefined,
                "right": undefined,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "start": undefined,
                "top": undefined,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "rgba(103, 80, 164, 1)",
                    "borderColor": "transparent",
                    "borderRadius": 20,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "flexBasis": "46%",
                    "margin": 5,
                    "minWidth": 64,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      false,
                      {
                        "borderRadius": 20,
                      },
                    ]
                  }
                  testID="button"
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "marginLeft": 12,
                            "marginRight": -4,
                          },
                          {
                            "marginLeft": 16,
                            "marginRight": -16,
                          },
                          false,
                        ]
                      }
                      testID="button-icon-container"
                    >
                      <Text
                        accessibilityElementsHidden={true}
                        allowFontScaling={false}
                        importantForAccessibility="no-hide-descendants"
                        pointerEvents="none"
                        selectable={false}
                        style={
                          [
                            {
                              "color": "rgba(255, 255, 255, 1)",
                              "fontSize": 18,
                            },
                            [
                              {
                                "lineHeight": 18,
                                "transform": [
                                  {
                                    "scaleX": 1,
                                  },
                                ],
                              },
                              {
                                "backgroundColor": "transparent",
                              },
                            ],
                            {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                      >
                        ó°
                      </Text>
                    </View>
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          {
                            "textAlign": "left",
                          },
                          {
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            {
                              "color": "rgba(255, 255, 255, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      mainMenu.points
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": 0,
        "end": undefined,
        "left": undefined,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 8,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 3,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        style={
          {
            "backgroundColor": "rgba(234, 221, 255, 1)",
            "borderRadius": 50,
            "margin": 16,
            "opacity": 1,
            "overflow": "hidden",
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
        testID="fab-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              undefined,
            ]
          }
          testID="fab"
        >
          <View
            pointerEvents="none"
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 16,
                  "height": 56,
                  "width": 56,
                },
              ]
            }
            testID="fab-content"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": [
                      {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(33, 0, 93, 1)",
                        "fontSize": 24,
                      },
                      [
                        {
                          "lineHeight": 24,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  ó°
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
